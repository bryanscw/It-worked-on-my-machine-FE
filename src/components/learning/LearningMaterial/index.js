import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import Loader from '../../common/Loader';
import { listLearningMaterials, selectLearningMaterialsListed, selectLearningMaterial } from '../../../redux/ducks/learningMaterials';
import './styles.css';

export class LearningMaterial extends Component {
    constructor(props) {
        super(props);

        props.listLearningMaterials();
    }

    render() {
        const {
            learningMaterialsListed,
            learningMaterial
        } = this.props;

        if (!learningMaterialsListed)
            return <Loader />;

        return (
            <Fragment>
                {
                    learningMaterial
                        ? <div className="card">
                            <div className="video-box card-img-top">
                                <div>
                                    <iframe
                                        title="Learning Material Video"
                                        src={learningMaterial.link}
                                        frameBorder="0"
                                        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                                        allowFullScreen />
                                </div>
                            </div>
                            <div className="card-body">
                                <h3 className="card-title">{learningMaterial.title}</h3>
                                <p className="card-text">{learningMaterial.description}</p>
                            </div>
                        </div>
                        : <p>No learning material found.</p>
                }
            </Fragment>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    const levelID = ownProps.level;

    return {
        learningMaterialsListed: selectLearningMaterialsListed(state),
        learningMaterial: selectLearningMaterial(state, levelID),
    }
};

const dispatchers = {
    listLearningMaterials,
};

export default connect(mapStateToProps, dispatchers)(LearningMaterial);
