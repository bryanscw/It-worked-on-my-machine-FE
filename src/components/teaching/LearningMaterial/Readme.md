```js
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap/dist/js/bootstrap.js';

var lmobj={
    title:"Basic Addition",
    description:"This is an interesting video",
    link:'https://www.youtube.com/embed/AuX7nPBqDts'
};
var editablebool=false;
<LearningMaterial editable={editablebool} learningMaterial={lmobj}/>
```