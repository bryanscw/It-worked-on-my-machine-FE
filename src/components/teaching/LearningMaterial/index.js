import React, { Component, Fragment } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEdit, faTrash } from '@fortawesome/free-solid-svg-icons';
import { connect } from 'react-redux';

import Loader from '../../common/Loader';
import { createLearningMaterial, retrieveLearningMaterials, updateLearningMaterial, deleteLearningMaterial } from '../../../redux/ducks/learningMaterials';
import './styles.css';

export class LearningMaterial extends Component {
    constructor(props) {
        super(props);

        props.retrieveLearningMaterials();
    }

    render() {
        const {
            learningMaterialsRetrieved,
            learningMaterial
        } = this.props;

        if (!learningMaterialsRetrieved)
            return <Loader />;

        return (
            <Fragment>
                {
                    learningMaterial
                        ? <div className="card">
                            <div className="video-box card-img-top">
                                <div>
                                    <iframe
                                        title="Learning Material Video"
                                        src={learningMaterial.link}
                                        frameBorder="0"
                                        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                                        allowFullScreen />
                                </div>
                            </div>
                            <div className="card-body">
                                <h3 className="card-title">{learningMaterial.title}</h3>
                                <p className="card-text">{learningMaterial.description}</p>
                                <button href="#" className="btn btn-success mr-2">
                                    <FontAwesomeIcon icon={faEdit} />
                                </button>
                                <button href="#" className="btn btn-danger">
                                    <FontAwesomeIcon icon={faTrash} />
                                </button>
                            </div>
                        </div>
                        : <p>No learning material found.</p>
                }
            </Fragment>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    const levelId = ownProps.level;

    return {
        learningMaterialsRetrieved: state.learningMaterialsReducer.learningMaterialsRetrieved,
        learningMaterial: state.learningMaterialsReducer.learningMaterials.find(learningMaterial => learningMaterial.level === levelId)
    }
};

const dispatchers = {
    createLearningMaterial,
    retrieveLearningMaterials,
    updateLearningMaterial,
    deleteLearningMaterial
};

export default connect(mapStateToProps, dispatchers)(LearningMaterial);
