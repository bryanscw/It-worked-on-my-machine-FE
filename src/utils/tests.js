import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    return <Provider store={store}>{component}</Provider>;
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return <Router history={history}>{component}</Router>
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    return renderWithRedux(renderWithRouter(component, route), initialState);
}