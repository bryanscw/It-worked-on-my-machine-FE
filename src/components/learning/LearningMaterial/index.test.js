import React from 'react'
import { cleanup, waitForElement } from '@testing-library/react'
import axiosMock from 'axios'

import LearningMaterial from './index.js';
import { renderWithReduxRouter } from '../../../utils/tests.js';

jest.mock('axios');

beforeEach(() => {
    axiosMock.get.mockResolvedValue({
        data: {
            content: [
                {
                    id: 1,
                    title: 'Basic Addition',
                    description: 'This is an interesting video',
                    link: 'https://www.youtube.com/embed/AQ7THUKx6Es',
                },
            ]
        }
    });
});

afterEach(cleanup);

it('should take a snapshot', async () => {
    const { asFragment, getByText } = renderWithReduxRouter(<LearningMaterial />);
    await waitForElement(() => getByText('Basic Addition'));
    expect(asFragment()).toMatchSnapshot();
})