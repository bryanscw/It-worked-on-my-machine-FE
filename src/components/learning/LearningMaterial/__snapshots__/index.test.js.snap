// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="video-box card-img-top"
    >
      <div>
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowfullscreen=""
          frameborder="0"
          src="https://www.youtube.com/embed/AQ7THUKx6Es"
          title="Learning Material Video"
        />
      </div>
    </div>
    <div
      class="card-body"
    >
      <h3
        class="card-title"
      >
        Basic Addition
      </h3>
      <p
        class="card-text"
      >
        This is an interesting video
      </p>
    </div>
  </div>
</DocumentFragment>
`;
